packages/kit/src/runtime/telemetry/record_span.js

```diff
@@ -0,0 +1,71 @@
+/** @import { Attributes, Span, Tracer } from '@opentelemetry/api' */
+import { HttpError, Redirect } from '../control.js';
+import { load_status_code } from './load_otel.js';
+import { noop_span } from './noop.js';
+
+/**
+ * @template T
+ * @param {Object} options
+ * @param {string} options.name
+ * @param {Tracer} options.tracer
+ * @param {Attributes} options.attributes
+ * @param {function(Span): Promise<T>} options.fn
+ * @returns {Promise<T>}
+ */
+export async function record_span({ name, tracer, attributes, fn }) {
+	const SpanStatusCode = await load_status_code();
+	if (SpanStatusCode === null) {
+		return fn(noop_span);
+	}
+
+	return tracer.startActiveSpan(name, { attributes }, async (span) => {
+		try {
+			const result = await fn(span);
+			span.end();
+			return result;
+		} catch (error) {
+			if (error instanceof HttpError) {
+				span.setAttributes({
+					[`${name}.result.type`]: 'known_error',
+					[`${name}.result.status`]: error.status,
+					[`${name}.result.message`]: error.body.message
+				});
+				span.recordException({
+					name: 'HttpError',
+					message: error.body.message
+				});
```

Comment:
does a `HttpError` also have a `stack`? If so, would be cool to also record the stack trace.

Reply:
Furthermore, I'm not 100% sure if this applies here but in our SDK, when we capture `HttpError`, we ignore 4xx errors because usually you don't want to record exceptions for bad requests, 404s, etc. Might be worth a thought if this is valuable or too noisy but happy to leave the call up to you!

https://github.com/getsentry/sentry-javascript/blob/8beaa4ea602d3517c272f07b8546325059bcfc21/packages/sveltekit/src/client/load.ts#L23-L38